navigation/src/main/java/no/nordicsemi/android/common/navigation/Navigator.kt

```diff
@@ -52,15 +54,20 @@ interface Navigator {
  * @param destination The current destination.
  */
 fun <A> SavedStateHandle.getStateFlow(destination: DestinationId<A, *>, initial: A?): StateFlow<A?> =
-    @Suppress("UNCHECKED_CAST")
-    getStateFlow<A?>(destination.name, initial)
+    getStateFlow(destination.name, initial)
+
+/**
+ * Returns the argument for the current destination.
+ *
+ * @param destination The current destination.
+ */
+fun <A> SavedStateHandle.getOrNull(destination: DestinationId<A?, *>): A? =
+    get(destination.name)
 
 /**
  * Returns the argument for the current destination.
  *
  * @param destination The current destination.
  */
-@Suppress("UNCHECKED_CAST")
-fun <A> SavedStateHandle.get(destination: DestinationId<A, *>): A? =
-    @Suppress("DEPRECATION")
-    get<Bundle?>(destination.name)?.get(destination.name) as? A
\ No newline at end of file
+fun <A> SavedStateHandle.get(destination: DestinationId<A & Any, *>): A =
+    get(destination.name) ?: error("Argument for ${destination.name} not found")
```

Comment:
I would write more specific e.g.: error("Shouldn't be used for nullable types").